site/server/app/com/fortysevendeg/exercises/services/interpreters/interpreters.scala

```diff
@@ -25,15 +30,42 @@ object ProdInterpreters extends InterpreterInstances {
 
   }
 
-  implicit def userOpsInterpreter: UserOp ~> Task = new (UserOp ~> Task) {
-
-    def apply[A](fa: UserOp[A]): Task[A] = ???
-
+  def userOpsInterpreter: UserOp ~> Task = new (UserOp ~> Task) {
+
+    def apply[A](fa: UserOp[A]): Task[A] = fa match {
+      case GetUsers()            ⇒ Task.delay(userService.all)
+      case GetUserByLogin(login) ⇒ Task.delay(userService.getUserByLogin(login))
+      case CreateUser(
```

Comment:
CreateUser should have been passed straight to the service as it may be decomposed there and applied to the query if needed via unapply